model_analyzer/config/generate/config_parameters.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, List, Optional, Tuple
+
+from model_analyzer.model_analyzer_exceptions import TritonModelAnalyzerException
+
+from .config_parameter import ConfigParameter, ParameterCategory, ParameterType
+
+
+class ConfigParameters:
+    """
+    Contains information about all configuration parameters the user wants to search
+    """
+
+    def __init__(self):
+        self._parameters: Dict[str, ConfigParameter] = {}
+
+    def add_parameter(
+        self,
+        name: str,
+        ptype: ParameterType,
+        category: ParameterCategory,
+        min_range: Optional[int] = None,
+        max_range: Optional[int] = None,
+        enumerated_list: List[Any] = [],
+    ):
+        self._check_for_illegal_input(category, min_range, max_range, enumerated_list)
+
+        self._parameters[name] = ConfigParameter(
+            ptype, category, min_range, max_range, enumerated_list
+        )
+
+    def get_parameter(self, name: str) -> ConfigParameter:
+        return self._parameters[name]
+
+    def get_type(self, name: str) -> ParameterType:
+        return self._parameters[name].ptype
+
+    def get_category(self, name: str) -> ParameterCategory:
+        return self._parameters[name].category
+
+    def get_range(self, name: str) -> Tuple[int, int]:
+        return (self._parameters[name].min_range, self._parameters[name].max_range)
+
+    def get_list(self, name: str) -> List[Any]:
+        return self._parameters[name].enumerated_list
```

Comment:
Should these be handled by ConfigParameters object? I feel like we just need to provide a single accessor to the underlying parameter by name and the users can use ConfigParameter object to get whatever they need with that parameter (since it's a dataclass).

Reply:
I'm not sure. Maybe this is overkill and never gets used, but I figured it couldn't hurt to add these.

Reply:
IMO I don't think it will be late to add these later, but I'll leave it up to you to make the final call.

Reply:
There in and tested at this point, so it'd be more work to rip them out. Going to leave it as is and merge.